src/main/java/name/webdizz/sonar/grammar/spellcheck/GrammarDictionaryLoader.java

```diff
@@ -1,33 +1,32 @@
 package name.webdizz.sonar.grammar.spellcheck;
 
+import com.swabunga.spell.engine.SpellDictionary;
+import com.swabunga.spell.engine.SpellDictionaryHashMap;
+import name.webdizz.sonar.grammar.GrammarPlugin;
+import name.webdizz.sonar.grammar.exceptions.UnableToLoadDictionary;
+import name.webdizz.sonar.grammar.utils.SpellCheckerUtil;
+import org.sonar.api.BatchExtension;
+import org.sonar.api.config.Settings;
+
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 
-import org.sonar.api.config.Settings;
-import name.webdizz.sonar.grammar.exceptions.UnableToLoadDictionary;
-import name.webdizz.sonar.grammar.utils.SpellCheckerUtil;
-
-import com.swabunga.spell.engine.SpellDictionary;
-import com.swabunga.spell.engine.SpellDictionaryHashMap;
-
-public class GrammarDictionaryLoader {
+public class GrammarDictionaryLoader implements BatchExtension {
 
     private final Lock locker = new ReentrantLock();
-    private Settings settings;
-    private AtomicReference<SpellDictionary> dictionary = new AtomicReference<>();
-    private String dictionaryPath = "/dict/english.0";
 
-    public GrammarDictionaryLoader() {
-    }
+    private AtomicReference<SpellDictionary> dictionary = new AtomicReference<>();
+    private String dictionaryPath;
 
-    public GrammarDictionaryLoader(final Settings settings) {
-        this.settings = settings;
+    public GrammarDictionaryLoader(Settings settings) {
+        dictionaryPath = settings.getString(GrammarPlugin.DICTIONARY);

```

Comment:
Could you please clarify, what do you mean by removing constant? Delete private String dictionaryPath and use like this : 
        try (InputStreamReader wordList = new InputStreamReader(SpellCheckerUtil.class.getResourceAsStream(settings.getString(GrammarPlugin.DICTIONARY)))) {